core/src/main/java/org/apache/carbondata/core/datastore/impl/FileFactory.java

```diff
@@ -81,16 +81,30 @@ public static FileReader getFileHolder(FileType fileType, Configuration configur
   }
 
   public static FileType getFileType(String path) {
-    String lowerPath = path.toLowerCase();
-    if (lowerPath.startsWith(CarbonCommonConstants.HDFSURL_PREFIX)) {
+    if (path.startsWith(CarbonCommonConstants.HDFSURL_PREFIX) || path.toLowerCase()
```

Comment:
I think this will do more object creations if the type is not HDFS because of many times `||` conditions and lower case conversions.  Please extract a private method and call first with actual string if it returns null then convert to lowercase and call again the same method. if it still returns null then return LOCAL

Reply:
Okay!!
I forgot remove lower case condition from first if check. I will extract to method